src/app/core/user/user-security/user-security.component.spec.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  ComponentFixture,
+  fakeAsync,
+  flush,
+  TestBed,
+  tick,
+} from "@angular/core/testing";
+
+import { UserSecurityComponent } from "./user-security.component";
+import { UserModule } from "../user.module";
+import { MockedTestingModule } from "../../../utils/mocked-testing.module";
+import { HttpClient, HttpErrorResponse } from "@angular/common/http";
+import { AuthService } from "../../session/auth/auth.service";
+import {
+  KeycloakAuthService,
+  Role,
+} from "../../session/auth/keycloak/keycloak-auth.service";
+import { of, throwError } from "rxjs";
+import { User } from "../user";
+
+describe("UserSecurityComponent", () => {
+  let component: UserSecurityComponent;
+  let fixture: ComponentFixture<UserSecurityComponent>;
+  let mockHttp: jasmine.SpyObj<HttpClient>;
+  const assignedRole: Role = {
+    id: "assigned-role",
+    name: "Assigned Role",
+    description: "this role is assigned to the user",
+  };
+  const notAssignedRole: Role = {
+    id: "not-assigned-role",
+    name: "Not Assigned Role",
+    description: "this role is not assigned to the user",
+  };
+  const user = { name: "test-user" } as User;
+  const keycloakUser = {
+    id: "userId",
+    email: "my@email.de",
+    roles: [assignedRole],
+  };
+
+  beforeEach(async () => {
+    mockHttp = jasmine.createSpyObj(["get", "put", "post"]);
+    mockHttp.get.and.returnValue(of([assignedRole, notAssignedRole]));
+    mockHttp.put.and.returnValue(of({}));
+    mockHttp.post.and.returnValue(of({}));
+
+    await TestBed.configureTestingModule({
+      imports: [UserModule, MockedTestingModule],
+      providers: [
+        { provide: AuthService, useValue: new KeycloakAuthService(mockHttp) },
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(UserSecurityComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it("should create", () => {
+    expect(component).toBeTruthy();
+  });
+
+  it("should load existing account data", fakeAsync(() => {
+    mockHttp.get.and.returnValue(of(keycloakUser));
+
+    component.onInitFromDynamicConfig({ entity: user });
+    tick();
+
+    expect(component.userId).toBe("userId");
+    expect(component.form.get("username").value).toBe(user.name);
```

Comment:
```suggestion
    expect(component.form.get("username")).toHaveValue(user.name);
```

Same with the two `expects` below.

Alternatively, you should also be able to check the whole form:

```
expect(component.form).toHaveValue({ username: user.name, email: ... });
```

with a potential `jasmine.objectContaining({...})` if the form has more values that you don't want to check